### PR TITLE
fix(distinct): Pmongo package to allow distinct method

### PR DESCRIPTION
**Pull request by sarweshmaharjan**
# Context
- **To Achieve**: Allow a Distinct method to be included in the pmongo.go 
--- 

## Description of the change

> Distinct method is present but not used in pmongo.go

### Type of change

- Add a Distinct method to pmongo.

### Details of the changes

- [x] New: Distinct method to be available in the pmongo package.

### CODE REVIEW

File: pkg/core/pmongo.go [hunk 1]
```diff
@@ -346,3 +346,37 @@ func (s *DBConnection) BulkWriteUpdate(ctx context.Context, collectionName strin
...
+//   if err != nil {
+//       log.Fatal(err)
+//   }
+//   fmt.Println(result)
+
+func (s *DBConnection) Unique(ctx context.Context, fieldName string, query interface{}, document Document) ([]interface{}, error) {
+	result, err := s.Collection(document.CollectionName()).Distinct(ctx, fieldName, query)
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			log.Printf("No documents found for query: %s\n", query)
```
[the remark this answers is not in the document]
Comment (sarweshmaharjan) [reply pkg/core/pmongo.go]: @abhyuday - Solved the issue.

File: pkg/core/pmongo.go [hunk 2]
```diff
@@ -346,3 +346,35 @@ func (s *DBConnection) BulkWriteUpdate(ctx context.Context, collectionName strin
...
+//   if err != nil {
+//       log.Fatal(err)
+//   }
+//   fmt.Println(result)
+
+func (s *DBConnection) Unique(ctx context.Context, fieldName string, query interface{}, document Document) ([]interface{}, error) {
+	result, err := s.Collection(document.CollectionName()).Distinct(ctx, fieldName, query)
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			return nil, err
```
Comment (asis012) [inline pkg/core/pmongo.go:375]: this extra check is also returning nil, err
anything specific to use this?

Comment (sarweshmaharjan) [reply pkg/core/pmongo.go:375]: @asis012 - Will remove it. I used it to cause the logs were present previously.

Comment (sarweshmaharjan) [reply pkg/core/pmongo.go]: @asis012 - Fixed

**Review by asis012 — APPROVED** [review]

**Review by abhyuday — APPROVED** [review]

**Pull request merged**